internal/validators/rbac_role_rule.go

```diff
@@ -0,0 +1,147 @@
+package validators
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v2"
+	"github.com/validator-labs/validator-plugin-azure/api/v1alpha1"
+	"github.com/validator-labs/validator-plugin-azure/internal/constants"
+	"github.com/validator-labs/validator-plugin-azure/internal/utils/azure"
+	vapi "github.com/validator-labs/validator/api/v1alpha1"
+	vapiconstants "github.com/validator-labs/validator/pkg/constants"
+	vapitypes "github.com/validator-labs/validator/pkg/types"
+)
+
+// roleAssignmentAPI contains methods that allow getting all role assignments for a scope and
+// optional filter.
+type roleAssignmentAPI interface {
+	GetRoleAssignmentsForScope(scope string, filter *string) ([]*armauthorization.RoleAssignment, error)
+}
+
+// roleDefinitionAPI contains methods that allow getting all the information we need for an existing
+// role definition.
+type roleDefinitionAPI interface {
+	GetByID(roleID string) (*armauthorization.RoleDefinition, error)
+}
+
+// RBACRoleRuleService reconciles RBAC role rules.
+type RBACRoleRuleService struct {
+	raAPI roleAssignmentAPI
+	rdAPI roleDefinitionAPI
+}
+
+// NewRBACRoleRuleService creates a new RBACRoleRuleService. Requires Azure client facades that
+// support, role assignments, and role definitions.
+func NewRBACRoleRuleService(raAPI roleAssignmentAPI, rdAPI roleDefinitionAPI) *RBACRoleRuleService {
+	return &RBACRoleRuleService{
+		raAPI: raAPI,
+		rdAPI: rdAPI,
+	}
+}
+
+// ReconcileRBACRoleRule reconciles an RBAC role rule.
+func (s *RBACRoleRuleService) ReconcileRBACRoleRule(rule v1alpha1.RBACRoleRule) (*vapitypes.ValidationRuleResult, error) {
+
+	// Build the default ValidationResult for this role assignment rule.
+	state := vapi.ValidationSucceeded
+	latestCondition := vapi.DefaultValidationCondition()
+	latestCondition.Failures = []string{}
+	latestCondition.Message = "All role assignments correct. Roles contain permissions and assigned to principal at scopes."
+	latestCondition.ValidationRule = fmt.Sprintf("%s-%s", vapiconstants.ValidationRulePrefix, rule.Name)
```

Comment:
please sanitize the name here using the new helper in validator core's string utils